transport.go

```diff
@@ -181,19 +198,39 @@ func (t *Transport) getClientConn(host, port string) (*clientConn, error) {
 }
 
 func (t *Transport) newClientConn(host, port, key string) (*clientConn, error) {
-	cfg := &tls.Config{
-		ServerName:         host,
-		NextProtos:         []string{NextProtoTLS},
-		InsecureSkipVerify: t.InsecureTLSDial,
-	}
-	tconn, err := tls.Dial("tcp", host+":"+port, cfg)
-	if err != nil {
-		return nil, err
+	cfg := t.TLSClientConfig
+	if cfg == nil {
+		cfg = &tls.Config{
+			ServerName: host,
+			NextProtos: []string{NextProtoTLS},
+		}
 	}
-	if err := tconn.Handshake(); err != nil {
-		return nil, err
+
+	var (
+		tconn *tls.Conn
+		err   error
+	)
+	if t.DialTLS != nil {
+		var conn net.Conn
+		conn, err = t.DialTLS("tcp", host+":"+port)
+		if err != nil {
+			return nil, err
+		}
+		var ok bool
+		tconn, ok = conn.(*tls.Conn)
+		if !ok {
+			// TODO(bmizerany): remove this check when we're ready
+			// to support non-TLS conns. I thought about making
+			// DialTLS return a *tls.Conn for now, but I wanted to
+			// keep the Transport interface as similar to
+			// net/http.Transport as possible as we progress.
+			panic("expected *tls.Conn")
```

Comment:
is there a reason to do the check with `ok` instead of just `tconn = conn.(*tls.Conn)` and just letting it panic if it's not a `*tls.Conn` ?